src/test/java/org/opensearch/flowframework/rest/FlowFrameworkSecureRestApiIT.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.rest;
+
+import org.opensearch.client.ResponseException;
+import org.opensearch.common.util.io.IOUtils;
+import org.opensearch.flowframework.FlowFrameworkRestTestCase;
+import org.opensearch.flowframework.TestHelpers;
+import org.opensearch.flowframework.model.Template;
+import org.junit.After;
+
+import java.io.IOException;
+
+public class FlowFrameworkSecureRestApiIT extends FlowFrameworkRestTestCase {
+
+    @After
+    public void tearDownSecureTests() throws IOException {
+        IOUtils.close(fullAccessClient(), readAccessClient());
+        deleteUser(FULL_ACCESS_USER);
+        deleteUser(READ_ACCESS_USER);
+    }
+
+    public void testCreateWorkflowWithReadAccess() throws Exception {
```

Comment:
should we add a happy case as well?

Reply:
Yes, Ill add more tests for those cases as well